webpack/ForemanTasks/Components/TasksTable/TasksTablePage.js

```diff
@@ -71,50 +63,74 @@ const TasksTablePage = ({
     parentTaskID,
   } = props;
   const tasksActions = {
-    cancelSelectedTasks: () => {
-      props.allRowsSelected
-        ? bulkCancelBySearch({ query: uriQuery, parentTaskID })
-        : bulkCancelById({
-            selected: getSelectedTasks(),
-            url,
-            parentTaskID,
-          });
+    [CANCEL_SELECTED_MODAL]: {
+      action: () => {
+        props.allRowsSelected
+          ? bulkCancelBySearch({ query: uriQuery, parentTaskID })
+          : bulkCancelById({
+              selected: getSelectedTasks(),
+              url,
+              parentTaskID,
+            });
+      },
+      actionText: 'cancel',
+      actionState: 'stopped',
     },
-    cancelTask: () => {
-      cancelTask({
-        taskId: clicked.taskId,
-        taskName: clicked.taskName,
-        url,
-        parentTaskID,
-      });
+    [CANCEL_MODAL]: {
+      action: () => {
+        cancelTask({
+          taskId: clicked.taskId,
+          taskName: clicked.taskName,
+          url,
+          parentTaskID,
+        });
+      },
+      actionText: 'cancel',
+      actionState: 'stopped',
     },
-    resumeSelectedTasks: () => {
-      props.allRowsSelected
-        ? bulkResumeBySearch({ query: uriQuery, parentTaskID })
-        : bulkResumeById({
-            selected: getSelectedTasks(),
-            url,
-            parentTaskID,
-          });
+
+    [RESUME_SELECTED_MODAL]: {
+      action: () => {
+        props.allRowsSelected
+          ? bulkResumeBySearch({ query: uriQuery, parentTaskID })
+          : bulkResumeById({
+              selected: getSelectedTasks(),
+              url,
+              parentTaskID,
+            });
+      },
+      actionText: 'resume',
+      actionState: 'running',
     },
-    resumeTask: () => {
-      resumeTask({
-        taskId: clicked.taskId,
-        taskName: clicked.taskName,
-        url,
-        parentTaskID,
-      });
+    [RESUME_MODAL]: {
+      action: () => {
+        resumeTask({
+          taskId: clicked.taskId,
+          taskName: clicked.taskName,
+          url,
+          parentTaskID,
+        });
+      },
+      actionText: 'resume',
+      actionState: 'running',
```

Comment:
it feels like something that should be part of redux,
for example when this action is triggered, these values should be updated in the store, and than consumed by the modal.

Reply:
What do you about making the confirm modal connected to redux? (with those fields and more)

Reply:
yeah I think it's a great idea, this way the `TasksTablePage` will become much simpler,
if you could move those action to be consumed by the modal it will also be awesome

Reply:
I think the actions depend on too much information from the page so I only moved the `UPDATE_MODAL` to the `ConfirmModal` new reducer. during the update the actionText and actionState get updated as well